File: lab7/frequency.pl
#!/usr/bin/perl -w
 $pattern = "$ARGV[0]";
 @files = glob("lyrics/*.txt");

foreach  $file (@files){
	$count = 0;
	$wordcount = 0;
	open (F,'<', $file) or die;
	while($line = <F>){
	
	$wordcount += (@words) = $line =~ m/[a-zA-Z]+/g;
	foreach  $word (@words){
			if ($word =~ m/^$pattern$/i){
				$count++;
			}
		}
	}
	
	$file =~ s/.txt//;
	$file =~ s/lyrics\///;
	$file =~ s/_/ /g;
	
	if ($wordcount == 0){print "file is $file"};
	
	printf "%4d/ %6d = %.9f %s\n",$count,$wordcount,($count/$wordcount), $file;
	close F;
}

File: lab7/frequency.py
#!/usr/bin/python 
import sys
import re
import glob


pattern = sys.argv[1].lower()

lyrics = glob.glob('lyrics/*.txt')
lyrics.sort()
for lyric in lyrics:
    wordCount = 0
    matches = 0
    F = open(lyric,"r")
    for line in F:
        words = re.findall('[a-zA-Z]+',line,flags=0)
        for word in words:
            if word.lower() == pattern:
                matches +=1
        wordCount += len(words);
    artist = re.sub('.txt','',lyric,count=0,flags=0)
    artist = re.sub('lyrics\/','',artist,count=0,flags=0)
    artist = re.sub("_"," ",artist,count=0,flags=0)
    
    frequency = float(matches)/float(wordCount)
    
    print "log(("+str(matches)+"/ "+str(wordCount)+") = ",'{:.9f}'.format(frequency),artist
    F.close()

